src/Plugin/WebformElement/CivicrmOptions.php

```diff
@@ -321,4 +321,33 @@ public function validateConfigurationForm(array &$form, FormStateInterface $form
     }
   }
 
+  /**
+   * @inheritDoc
+   */
+  protected function format($type, array &$element, WebformSubmissionInterface $webform_submission, array $options = []) {
+    $value = parent::format($type, $element, $webform_submission, $options);
+    $format = $this->getItemFormat($element);
+    if (!str_ends_with($element['#form_key'], '_address_state_province_id')) {
+      return $value;
+    }
+    if ($type === 'Text') {
+      $state_id = $value;
+    }
+    else {
+      $state_id = $value['#plain_text'] ?? NULL;
```

Comment:
If you're sending an email with `state_province_id` fields, and the email is in HTML format, `$value['#plain_text']` doesn't exist.  I this we want `$state_id = $value['#plain_text'] ?? $value['#markup'] ?? NULL;`.

Reply:
@MegaphoneJon If i use the default body having`[webform_submission:values]` token to render all submission, it does work for me after the current PR. (Screenshot from email)

![image](https://github.com/colemanw/webform_civicrm/assets/5929648/feede4fd-598f-4006-b346-d515382c269d)

If i use a custom body with field tokens `[webform_submission:values: civicrm_1_contact_1_address_state_province_id]`, it still works and renders the state correctly in email

![image](https://github.com/colemanw/webform_civicrm/assets/5929648/cf3b79ba-6a74-4a86-b350-de3550374bff)

The debug value of $text is Html & $value does have `#plain_text` in this function -

![image](https://github.com/colemanw/webform_civicrm/assets/5929648/25ebadaf-a522-47ca-bf46-2f80de811fb3)

I'm fine to include your change if `$value['#markup']` works for you. But curious to know your usecase since we have a [similar implementation](https://github.com/colemanw/webform_civicrm/blob/6.x/src/Plugin/WebformElement/CivicrmContact.php#L601) for Existing Contact, so i think `[webform_submission:values:civicrm_1_contact_1_contact_existing]` token might break for you too?

Reply:
Hi Jitendra,

That's not what I'm seeing.  Attached is a webform export that displays the state/province as an ID.

![Selection_2143](https://github.com/colemanw/webform_civicrm/assets/1796012/b4165ef8-3a1c-420d-b765-a7af3c1d5dc2)

[state id issue.yml.txt](https://github.com/colemanw/webform_civicrm/files/14240652/state.id.issue.yml.txt)

Reply:
Hmm, the attached webform still works fine on my end. Both Preview and email has state label instead of an id. Also, the debug values are same.

<img width="482" alt="image" src="https://github.com/colemanw/webform_civicrm/assets/5929648/967502d4-e843-48d6-aa4a-d0605a715494">

I believe I'll proceed with your suggestion since it appears to be a safe addition and works on your end :)